### PR TITLE
LinePlot

### PR DESCRIPTION
**Pull request by freddyaboulton**
# Description
Addition of `gr.LinePlot` with api similar to `gr.ScatterPlot`. 

```python
gr.LinePlot(value=data,
                    x="date", y="downloads",
                    tooltip=['date', 'downloads'],
                    title=f"Pypi downloads of {lib} over last {time}",
                    overlay_point=True, height=400, width=900)
```

line_plot_demo

To test, run the unit tests and run native_plots demo. You can check it out in the deployed space once spaces start to build again.

I also added the following to both LinePlot and ScatterPlot:

- Ability to set x and y axis limits
- Ability to move the position of the legend

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added a short summary of my change to the CHANGELOG.md
- [x] My code follows the style guidelines of this project
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: gradio/components.py [hunk 1]
```diff
@@ -4046,6 +4046,22 @@ def style(self, container: Optional[bool] = None):
         )
 
 
+class AltairPlot:
```
Comment (freddyaboulton) [inline gradio/components.py:4049]: I expect the number of methods in `AltairPlot` to increase as we add more plots and find more common patterns that can be abstracted out.

For the time being I don't want to invest too much time in coming up with a fancy abstraction as the plotting code will be edited a lot over the coming months.

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,7 +33,7 @@ No changes to highlight.
 No changes to highlight.
 
 ## Full Changelog:
-No changes to highlight.
+* Added the ability to set `x_lim`, `y_lim` and legend positions for `gr.ScatterPlot` by  [@freddyaboulton](https://github.com/freddyaboulton) in [PR 2807](https://github.com/gradio-app/gradio/pull/2807)
```
Comment (abidlabs) [inline CHANGELOG.md:36]: Thank you!

File: gradio/components.py [hunk 3]
```diff
@@ -4181,10 +4218,15 @@ def update(
             color_legend_title: The title given to the color legend. By default, uses the value of color parameter.
             size_legend_title: The title given to the size legend. By default, uses the value of the size parameter.
             shape_legend_title: The title given to the shape legend. By default, uses the value of the shape parameter.
+            color_legend_position: The position of the color legend. If 'none', omit the legend. For other valid position values see: https://vega.github.io/vega/docs/legends/#orientation.
+            size_legend_position: The position of the size legend. If 'none', omit the legend. For other valid position values see: https://vega.github.io/vega/docs/legends/#orientation.
+            shape_legend_position: The position of the shape legend. If 'none', omit the legend. For other valid position values see: https://vega.github.io/vega/docs/legends/#orientation.
             height: The height of the plot in pixels.
             width: The width of the plot in pixels.
-            caption: The (optional) caption to display below the plot.
+            x_lim: A tuple or list containing the limits for the x-axis.
```
Comment (abidlabs) [inline gradio/components.py:4226]:
```suggestion
            x_lim: A tuple or list containing the limits for the x-axis, specified as (x_min, x_max).
```

File: gradio/components.py [hunk 4]
```diff
@@ -4181,10 +4218,15 @@ def update(
             color_legend_title: The title given to the color legend. By default, uses the value of color parameter.
             size_legend_title: The title given to the size legend. By default, uses the value of the size parameter.
             shape_legend_title: The title given to the shape legend. By default, uses the value of the shape parameter.
+            color_legend_position: The position of the color legend. If 'none', omit the legend. For other valid position values see: https://vega.github.io/vega/docs/legends/#orientation.
+            size_legend_position: The position of the size legend. If 'none', omit the legend. For other valid position values see: https://vega.github.io/vega/docs/legends/#orientation.
+            shape_legend_position: The position of the shape legend. If 'none', omit the legend. For other valid position values see: https://vega.github.io/vega/docs/legends/#orientation.
```
Comment (abidlabs) [inline gradio/components.py:4223]: Just to clarify, what does "If 'none', omit the legend" mean? if *any* of these `None`, the legend is omitted?

File: gradio/components.py [hunk 5]
```diff
@@ -4181,10 +4218,15 @@ def update(
...
             height: The height of the plot in pixels.
             width: The width of the plot in pixels.
-            caption: The (optional) caption to display below the plot.
+            x_lim: A tuple or list containing the limits for the x-axis.
+            y_lim: A tuple of list containing the limits for the y-axis.
```
Comment (abidlabs) [inline gradio/components.py:4227]:
```suggestion
            y_lim: A tuple of list containing the limits for the y-axis, specified as (y_min, y_max).
```

File: gradio/components.py [hunk 6]
```diff
@@ -4323,14 +4389,352 @@ def postprocess(self, y: pd.DataFrame | Dict | None) -> Dict[str, str] | None:
...
+            overlay_point: Whether to draw a point on the line for each (x, y) coordinate pair.
+            title: The title to display on top of the chart.
+            tooltip: The column (or list of columns) to display on the tooltip when a user hovers a point on the plot.
+            x_title: The title given to the x axis. By default, uses the value of the x parameter.
+            y_title: The title given to the y axis. By default, uses the value of the y parameter.
+            color_legend_title: The title given to the color legend. By default, uses the value of color parameter.
+            stroke_dash_legend_title: The title given to the stroke_dash legend. By default, uses the value of the stroke_dash parameter.
+            color_legend_position: The position of the color legend. If 'none', omit the legend. For other valid position values see: https://vega.github.io/vega/docs/legends/#orientation.
+            stroke_dash_legend_position: The position of the stoke_dash legend. If 'none', omit the legend. For other valid position values see: https://vega.github.io/vega/docs/legends/#orientation.
```
Comment (abidlabs) [inline gradio/components.py:4460]: Same question as above about the Scatterplot

File: gradio/components.py [hunk 7]
```diff
@@ -4323,14 +4389,352 @@ def postprocess(self, y: pd.DataFrame | Dict | None) -> Dict[str, str] | None:
...
+            height: The height of the plot in pixels.
+            width: The width of the plot in pixels.
+            x_lim: A tuple or list containing the limits for the x-axis.
+            y_lim: A tuple of list containing the limits for the y-axis.
```
Comment (abidlabs) [inline gradio/components.py:4464]:
```suggestion
            x_lim: A tuple or list containing the limits for the x-axis, specified as (x_min, x_max).
            y_lim: A tuple of list containing the limits for the y-axis, specified as (y_min, y_max).
```

**Comment by abidlabs** [comment]
I ran the `native_plots` demo and I'm getting this error:

```
altair.utils.schemapi.SchemaValidationError: Invalid specification

        altair.vegalite.v4.schema.channels.SizeValue, validating 'additionalProperties'

        Additional properties are not allowed ('selection' was unexpected)
```

I think it's because I'm using an older version of `altair`, let me check

**Review by abidlabs — APPROVED** [review]
Tested and everything works great, as long as I have `altair>=4.2.0`. Fantastic PR @freddyaboulton!

**Comment by freddyaboulton** [comment]
Thank you for the review @abidlabs !

**Pull request merged**